### PR TITLE
Merges `main` into `snarkeling` branch

### PR DESCRIPTION
**Pull request by deuszx**
# Description

Merge conflicts in:
* `Cargo.toml`s and `Cargo.lock`s
* associated event type in pallets changed from` type Event` to `type RuntimeEvent`
* Due to `Weights v2` changes we had to update the weight template in `.maintain/` folder and rerun the command generating `pallet/snarcos/src/weights.rs`

## Type of change

Merge `main` into `snarkeling`.

### CODE REVIEW

File: pallets/snarcos/src/weights.rs [hunk 1]
```diff
@@ -27,270 +27,255 @@ use sp_std::marker::PhantomData;
...
-            .max(<I as BenchmarkInfo>::verify_marlin_linear_equation())
-            .max(<I as BenchmarkInfo>::verify_marlin_merkle_tree_8())
-            .max(<I as BenchmarkInfo>::verify_marlin_merkle_tree_64())
-            .max(<I as BenchmarkInfo>::verify_marlin_merkle_tree_1024())
-    }
-
-    // TODO: compile results from benchmarks
+        fn store_key(key_length: u32) -> Weight {
+                       <I as BenchmarkInfo>::store_key(key_length)
+                   }
```
Comment (pmikolajczyk41) [inline pallets/snarcos/src/weights.rs:40]: pretty weird indendting

**Review by pmikolajczyk41 — APPROVED** [review]

File: benches/payout-stakers/Cargo.lock [hunk 2]
```diff
@@ -3708,19 +4283,32 @@ version = "0.4.0"
...
+checksum = "5a3e1820f08b8513f676f7ab6c1f99ff312fb97b553d30ff4dd86f9f15728aa7"
+dependencies = [
+ "windows_aarch64_msvc 0.36.1",
+ "windows_i686_gnu 0.36.1",
+ "windows_i686_msvc 0.36.1",
+ "windows_x86_64_gnu 0.36.1",
+ "windows_x86_64_msvc 0.36.1",
+]
+
 [[package]]
```
Comment (fbielejec) [inline benches/payout-stakers/Cargo.lock:4299]: CI failure here when parsing

Comment (deuszx) [reply benches/payout-stakers/Cargo.lock:4285]: This should be now fixed.

**Comment by pmikolajczyk41** [comment]
oh, I forgot to add: please find in `bin/runtime/lib.rs` a line with `let whitelist: Vec<TrackedStorageKey> = AllPalletsWithSystem::whitelisted_storage_keys();` and use it :)

**Pull request merged**